gui/wxpython/projpicker_gui/wxwidgets.py

```diff
@@ -0,0 +1,798 @@
+                lambda data: wx.Image(io.BytesIO(data)),
+                lambda image, tile, x, y: image.Paste(tile, x, y),
+                lambda tile, dz: tile.Scale(tile.Width*2**dz,
+                                            tile.Height*2**dz),
+                map_canvas_width, map_canvas_height,
+                lat, lon, zoom)
+
+        self.map_canvas.Bind(wx.EVT_LEFT_DOWN, self.on_grab)
+        self.map_canvas.Bind(wx.EVT_LEFT_UP, self.on_draw)
+        self.map_canvas.Bind(wx.EVT_LEFT_DCLICK, self.on_complete_drawing)
+        self.map_canvas.Bind(wx.EVT_RIGHT_UP, self.on_cancel_drawing)
+        self.map_canvas.Bind(wx.EVT_RIGHT_DCLICK, self.on_clear_drawing)
+        self.map_canvas.Bind(wx.EVT_MOTION, self.on_move)
+        self.map_canvas.Bind(wx.EVT_MOUSEWHEEL, self.on_zoom)
+        self.map_canvas.Bind(wx.EVT_SIZE, self.on_resize)
+        self.map_canvas.Bind(wx.EVT_PAINT, self.on_paint)
+        map_box.Add(self.map_canvas, 1, wx.EXPAND)
+
+        #######################
+        # label for coordinates
+
+        # label=" "*40 hack to reserve enough space for coordinates
+        self.coor_label = wx.StaticText(self, label=" "*40)
+        # to avoid redrawing the entire map just to refresh coordinates; use a
+        # vertical box sizer for right alignment, which is not allowed with a
+        # horizontal box sizer
+        self.coor_label.box = wx.BoxSizer(wx.VERTICAL)
+        self.coor_label.box.Add(self.coor_label, 0, wx.ALIGN_RIGHT)
+        map_box.Add(self.coor_label.box, 0, wx.ALIGN_RIGHT)
+
+        main_box.Add(map_box, 1, wx.EXPAND)
+
+        ####################
+        # bottom/right frame
+        bottom_box = wx.BoxSizer(wx.HORIZONTAL)
+
+        mono_font = wx.Font(9, wx.FONTFAMILY_TELETYPE, wx.FONTSTYLE_NORMAL,
+                            wx.FONTWEIGHT_NORMAL)
+
+        if self.layout == "full":
+            ###################
+            # bottom-left frame
+            bottom_left_box = wx.BoxSizer(wx.VERTICAL)
+
+            # list of CRSs
+            self.crs_list = wx.ListCtrl(self, style=wx.LC_REPORT |
+                                                    wx.LC_SINGLE_SEL)
+
+            self.crs_list.AppendColumn("ID")
+            self.crs_list.AppendColumn("Name")
+            self.crs_list.SetColumnWidth(0, 110)
+            self.crs_list.SetColumnWidth(1, wx.LIST_AUTOSIZE_USEHEADER)
+            bottom_left_box.Add(self.crs_list, 1, wx.EXPAND)
+
+            self.crs_list.Bind(wx.EVT_LIST_ITEM_SELECTED, self.on_select_crs)
+
+            # add text for search later to get the button height
+
+            bottom_box.Add(bottom_left_box, 1, wx.EXPAND)
+
+            ####################
+            # bottom-right frame
+            self.notebook = wx.Notebook(self)
+
+            # query tab
+            self.query_panel = wx.Panel(self.notebook)
+            self.query_panel.page = self.notebook.PageCount
+            self.notebook.AddPage(self.query_panel, "Query")
```

Comment:
The user-visible text should be translatable: `_("...")`

Reply:
Done in 67c3047